datarobot_batch_scoring/writer.py

```diff
@@ -397,30 +397,26 @@ def format_result_data(self, result, batch):
         skip_row_id = self.ctx.skip_row_id
         fast_mode = self.ctx.fast_mode
         input_delimiter = self.ctx.dialect.delimiter
-
-        predictions = result['predictions']
-        if result['task'] == TargetType.BINARY:
-            sorted_classes = list(
-                sorted(predictions[0]['class_probabilities'].keys()))
-            out_fields = ['row_id'] + sorted_classes
-            if pred_name is not None:
-                sorted_classes = [sorted_classes[-1]]
-                out_fields = ['row_id'] + [pred_name]
-            pred = [[p['row_id'] + batch.id] +
-                    [p['class_probabilities'][c] for c in
-                     sorted_classes]
-                    for p in
-                    sorted(predictions,
-                           key=operator.itemgetter('row_id'))]
-        elif result['task'] == TargetType.REGRESSION:
-            pred = [[p['row_id'] + batch.id, p['prediction']]
-                    for p in
-                    sorted(predictions,
-                           key=operator.itemgetter('row_id'))]
-            out_fields = ['row_id', pred_name if pred_name else '']
+        single_row = result[0]
+        fields = sorted(record['label']
+                        for record in single_row['predictionValues'])
+
+        if pred_name is not None:
+            out_fields = ['row_id', pred_name]
+            if len(fields) == 2 and single_row['prediction'] in fields:
```

Comment:
# batch scoring returns only positive class for binary tasks if 
            # user specified pred_name option, so we eliminate negative 
            # result